### PR TITLE
feature/material editor

### PR DESCRIPTION
**Pull request by EnriquePuyol**
Material Editor added inside the inspector.
Normal map added in the material editor
Save & Load edited for the Normal map

### CODE REVIEW

File: Source/MaterialEditor.cpp [hunk 1]
```diff
@@ -22,94 +22,79 @@ MaterialEditor::~MaterialEditor()
...
-			else
-			{
-				previous = new Material(*material); //Save changes if cancel
-			}
-		}
-		ImGui::SetNextWindowSizeConstraints(ImVec2(200, 200), ImVec2(500, 500));
+		return;
+	}
+
+	SetCurrentTextures();
```
Comment (AlbertVVila) [inline Source/MaterialEditor.cpp]: SetCurrentTextures is called in every loop, not only the first time

File: Source/MaterialEditor.cpp [hunk 2]
```diff
@@ -22,94 +22,79 @@ MaterialEditor::~MaterialEditor()
...
-			TextureSelector((unsigned)TextureType::EMISSIVE, current_emissive);
-			ImGui::Separator();
-		}
-		if (ImGui::Button("Save", ImVec2(120, 0))) {
-			material->Save();
-			CleanUp();
-		}
-		ImGui::SetItemDefaultFocus();
-		ImGui::SameLine();
-		if (ImGui::Button("Cancel", ImVec2(120, 0)))
```
Comment (AlbertVVila) [inline Source/MaterialEditor.cpp]: You can no longer cancel material changes?

Comment (AlbertVVila) [reply Source/MaterialEditor.cpp]: Save material not longer working?

File: Source/MaterialEditor.cpp [hunk 3]
```diff
@@ -57,40 +57,41 @@ void MaterialEditor::Draw()
...
 		ImGui::Separator();
 	}
 	if (ImGui::CollapsingHeader("Normal"))
 	{
-		TextureSelector((unsigned)TextureType::NORMAL, current_normal);
+		TextureSelector((unsigned)TextureType::NORMAL, current_normal, 4);
 	}
 
-	ImGui::SetCursorPosX(ImGui::GetWindowWidth()/2 - ImGui::CalcTextSize("SAVE").x /2);
-	if (ImGui::Button("SAVE"))
```
Comment (AlbertVVila) [inline Source/MaterialEditor.cpp]: Why save changes disappeared?

**Comment by EnriquePuyol** [comment]
Take a look at the state of:
-Creating a new material (also try to repeat the name of one that already exists)
- Editing a material (not renaming pls) + must be auto saved

**Pull request closed without merging**